vpdq/cpp/regtest.py

```diff
@@ -72,91 +74,114 @@ def get_argparse() -> argparse.ArgumentParser:
     return ap
 
 
-def dir_path(string):
-    if os.path.isdir(string):
-        return string
-    else:
-        raise argparse.ArgumentTypeError(f"readable_dir: {string} is not a valid path")
+def main():
+    hashVideoExecutable = EXEC_DIR / "vpdq-hash-video"
+    matchHashesExecutable = EXEC_DIR / "match-hashes-byline"
 
+    if not hashVideoExecutable.exists() or not matchHashesExecutable.exists():
+        print(
+            "Error: Hashing executable/s not found. Build vpdq before running regtest."
+        )
+        sys.exit(1)
 
-def main():
     ap = get_argparse()
     args = ap.parse_args()
     inputVideoFolder = args.inputVideoFolder
     outputHashFolder = args.outputHashFolder
-    ffmpegPath = args.ffmpegPath
     secondsPerHash = str(args.secondsPerHash)
     downsampleFrameDimension = str(args.downsampleFrameDimension)
-    verbose = args.verbose
-    # TODO: Add more general options for other video encodings.
-    for file in os.listdir(inputVideoFolder):
-        if file.endswith(".mp4"):
-            if verbose:
-                subprocess.call(
-                    [
-                        "./build/vpdq-hash-video",
-                        "-v",
-                        "-f",
-                        ffmpegPath,
-                        "-r",
-                        secondsPerHash,
-                        "-d",
-                        outputHashFolder,
-                        "-s",
-                        downsampleFrameDimension,
-                        "-i",
-                        inputVideoFolder + "/" + file,
-                    ]
-                )
-            else:
-                subprocess.call(
-                    [
-                        "./build/vpdq-hash-video",
-                        "-f",
-                        ffmpegPath,
-                        "-r",
-                        secondsPerHash,
-                        "-d",
-                        outputHashFolder,
-                        "-s",
-                        downsampleFrameDimension,
-                        "-i",
-                        inputVideoFolder + "/" + file,
-                    ]
-                )
-
-    cdir = os.getcwd()
-    pdir = os.path.dirname(cdir)
-    sample = pdir + ("/sample-hashes")
-    output = outputHashFolder
     distanceTolerance = str(args.matchDistanceTolerance)
     qualityTolerance = str(args.qualityTolerance)
-    for file in os.listdir(sample):
-        if file.endswith(".txt"):
-            print("\nMatching File " + file)
-            sampleFile = f"{sample}/{file}"
-            outputFile = f"{output}/{file}"
+    verbose = args.verbose
+
+    if not inputVideoFolder.exists():
+        raise argparse.ArgumentTypeError(
```

Comment:
nit: This is a misplaced ArgumentTypeError that will turn into a full stack trace. You can embed this into parsing for nice error messages by turning this into a function like

```
def _validate_path(string_path: str) -> Path:
   ret = Path(string_path)
   if not ret.exists():
      raise argparse.ArgumentTypeError("path does not exist")
   return ret

...
     ...
     type=_validate_path,
   )
```